### PR TITLE
[action] legacy tx compatible with version = 0

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
When we first enable web3 format tx (at around Nov 2021), `version = 0` is used, all such tx are legacy tx type. So we should treat `version = 0` as legacy tx to maintain compatibility

Since web3 format tx's hash does not depend on the value of `version`, this is a safe change only to be compatible with past data/blocks.

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/tx_legacy.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ var _ TxCommonInternal = (*LegacyTx)(nil)
 
 // LegacyTx is the legacy transaction
 type LegacyTx struct {
+	version  uint32
```
Comment (envestcc) [inline action/tx_legacy.go:21]: we can define a new Web3LegacyTx to handle this

File: action/tx_access_list.go [hunk 2]
```diff
@@ -148,23 +148,19 @@ func fromAccessListProto(list []*iotextypes.AccessTuple) types.AccessList {
...
 }
 
-func fromProtoAccessListTx(pb *iotextypes.ActionCore) (*AccessListTx, error) {
-	var tx AccessListTx
-	tx.nonce = pb.GetNonce()
-	tx.gasLimit = pb.GetGasLimit()
-	tx.chainID = pb.GetChainID()
-	tx.gasPrice = &big.Int{}
-
+func (tx *AccessListTx) fromProto(pb *iotextypes.ActionCore) error {
```
Comment (dustinxie) [inline action/tx_access_list.go:151]: refactor to make it consistent with other tx types

File: action/tx_access_list_test.go [hunk 3]
```diff
@@ -48,7 +48,15 @@ func TestAccessListTx(t *testing.T) {
...
 		pb := iotextypes.ActionCore{}
 		r.NoError(proto.Unmarshal(b, &pb))
-		r.Equal(tx, MustNoErrorV(fromProtoAccessListTx(&pb)))
+		tx1 := &AccessListTx{
```
Comment (dustinxie) [inline action/tx_access_list_test.go:51]: use dirty values to make sure `fromProto()` sets every field

File: action/tx_access_list.go [hunk 4]
```diff
@@ -148,23 +148,19 @@ func fromAccessListProto(list []*iotextypes.AccessTuple) types.AccessList {
...
 	}
-	if acl := pb.GetAccessList(); len(acl) > 0 {
-		tx.accessList = fromAccessListProto(acl)
-	}
-	return &tx, nil
+	tx.chainID = pb.GetChainID()
+	tx.nonce = pb.GetNonce()
+	tx.gasLimit = pb.GetGasLimit()
+	tx.gasPrice = &gasPrice
+	tx.accessList = fromAccessListProto(pb.GetAccessList())
```
Comment (dustinxie) [inline action/tx_access_list.go:162]: `fromAccessListProto()` returns `nil` for nil or empty slice, so can simplify

File: action/tx_access_list_test.go (hunk 3 above)
Comment (dustinxie) [inline action/tx_access_list_test.go:51]: use dirty values to make sure `fromProto()` sets every field

File: action/tx_blob.go [hunk 5]
```diff
@@ -194,7 +190,7 @@ func (tx *BlobTx) fromProto(pb *iotextypes.ActionCore) error {
 	tx.chainID = pb.GetChainID()
 	tx.gasFeeCap = feeCap
 	tx.gasTipCap = tipCap
-	tx.accessList = acl
+	tx.accessList = fromAccessListProto(pb.GetAccessList())
```
Comment (dustinxie) [inline action/tx_blob.go:193]: `fromAccessListProto()` returns `nil` for nil or empty slice, so can simplify

File: action/tx_legacy.go (hunk 1 above)
Comment (dustinxie) [reply action/tx_legacy.go]: updated

**Review by envestcc — APPROVED** [review]

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**